feat: migrate dag runs in batches

m1racoli opened the pull request:
We're adding an additional input for the desired batch size and adjust the migration logic to call itself recursively until all DAG runs are migrated.

m1racoli committed:
feat: migrate dag runs in batches

We're adding an additional input for the desired batch size and adjust
the migration logic to call itself recursively until all DAG runs
are migrated.

fritz-astronomer reviewed (approved):
❗ 
I don't have time to review _too_ hard today but this looks solid and is a real cool feature! 
If you gave it some good testing, 👍 from me. 
Nothing stands out

m1racoli committed:
fix: ensure we don't migrate more than the limit

If the batch size would result in a larger result than the limit allows
then we need to adjust the batch size accordingly.

m1racoli commented:
Found one small bug. Now it looks all good. Will wait until next week. Happy Friday 🍻
